FrontEnd/src/HomePage/RecommendationDB.js

```diff
@@ -40,87 +81,130 @@ const addToFavoriteDocs = async(userUID, favorited, post) =>{
       likedAt: new Date(),
     });
   }
-}
+};
 
-const isLiked = (userUID, postID, callback) =>{
-  const favoritesRef = collection(database, DATABASE_FOLDER_NAME, userUID, FAVORITES_COLLECTION)
+const isLiked = (userUID, postID, callback) => {
+  const favoritesRef = collection(
+    database,
+    DATABASE_FOLDER_NAME,
+    userUID,
+    FAVORITES_COLLECTION
+  );
   const q = query(favoritesRef, where("post.postID", "==", postID));
   const unsubscribe = onSnapshot(q, (querySnapshot) => {
     callback(!querySnapshot.empty);
-});
+  });
 
-return unsubscribe
-}
+  return unsubscribe;
+};
 
-const postsFromFavorites = async(userUID) =>{
-  try{
-    const favoritesRef = collection(database, DATABASE_FOLDER_NAME, userUID, FAVORITES_COLLECTION)
-    const favoritePosts = await getDocs(favoritesRef)
-    const favoriteVendorsID = []
+const postsFromFavorites = async (userUID) => {
+  try {
+    const favoritesRef = collection(
+      database,
+      DATABASE_FOLDER_NAME,
+      userUID,
+      FAVORITES_COLLECTION
+    );
+    const favoritePosts = await getDocs(favoritesRef);
+    const favoriteVendorsID = [];
 
-    for (const favoriteDoc of favoritePosts.docs){
-      const vendorID = favoriteDoc.data().userId
-      if (vendorID !== null){
-        favoriteVendorsID.push(vendorID)
+    for (const favoriteDoc of favoritePosts.docs) {
+      const vendorID = favoriteDoc.data().post.vendorUID;
+      if (vendorID !== null) {
+        favoriteVendorsID.push(vendorID);
       }
+    }
+
+    return favoriteVendorsID;
```

Comment:
In `postsFromFavorites`, why do we return `favoriteVendorsID`? Shouldn't it be the ids of the favorited posts?

Reply:
I want to use it to suggest more posts from the vendor